cui.cpp

```diff
@@ -215,6 +216,18 @@ void exit_ui ()
 
 void ui_tick ()
 {
+	fd_set fds;
+	
+	int maxfd = 0; // stdin
+
+	FD_ZERO(&fds);
+	FD_SET(0, &fds);
+	int status = select(maxfd + 1, &fds, 0, 0, 0);
```

Comment:
The docs suggest that using `NULL` as a timeout will do a blocking read, so I'd suppose this would block and prevent `dp_dispatch` to be run periodially in the main thread - but running the code suggests this is not the case. 

I don't have time to check out what's going on right now - could you clarify?